### PR TITLE
lib: multicell_location: Add support for nRF Cloud location service

### PR DESCRIPTION
**Pull request by simensrostad**
Add support for nRF Cloud location service.

Rebased ontop of https://github.com/nrfconnect/sdk-nrf/pull/4665.

CIA-329

### CODE REVIEW

File: lib/multicell_location/services/nrf_cloud_integration.c [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+			  "&mnc=%u"					\
+			  "&tac=%u"					\
+			  "&eci=%u"					\
+			  "&format=json"
+
+BUILD_ASSERT(sizeof(HOSTNAME) > 1, "Hostname must be configured");
+
+static char body[HTTP_BODY_SIZE];
+
+static const char tls_certificate[] =
```
Comment (rlubos) [inline lib/multicell_location/services/nrf_cloud_integration.c:51]: Would be good to add some info on the certificate origin, same as for other providers.

**Review by rlubos — COMMENTED** [review]
Just a small suggestion.

File: lib/multicell_location/services/nrf_cloud_integration.c (hunk 1 above)
Comment (simensrostad) [reply lib/multicell_location/services/nrf_cloud_integration.c:51]: Yeah. Agree. @jtguggedal will continue with this PR.

**Comment by jtguggedal** [comment]
The PR has been updated with documentation for the added nRF Cloud service and some fixes

File: lib/multicell_location/services/nrf_cloud_integration.c (hunk 1 above)
Comment (jtguggedal) [reply lib/multicell_location/services/nrf_cloud_integration.c:54]: Certificate info has been added

**Comment by jtguggedal** [comment]
Rebased after #4665 was merged and removed DNM label

**Review by umapraseeda — APPROVED** [review]

**Review by jhn-nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**